File: src/main/webapp/js/app/master/sender.service.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

(function() {

	var services = angular.module('je-master-sender-service', [ 'ngResource' ]);

	
	services.factory('SenderListFactory', [ '$resource', function($resource) {

		return $resource('service/master/sender', {}, {
			query : {
				method : 'GET',
				isArray : false
			},
			create : {
				method : 'POST'
			}
		});

	} ]);
	
	services.factory('SenderEditFactory', [ '$resource', function($resource) {

		return $resource('service/master/sender/:id', {}, {
			show : {
				method : 'GET',
				params: {id: '@id'}
			},
			update : {
				method : 'PUT',
				params :{id: '@id'}
			},
			remove : {
				method : 'DELETE',
				params :{id: '@id'}
			}
		});

	} ]);
	
})();

File: src/main/java/com/je/spring/rest/master/service/SenderService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.je.spring.rest.master.service;

import com.je.spring.rest.master.model.Sender;
import java.util.List;
/**
 *
 * @author Prasetyo
 */
public interface SenderService {
    public List<Sender> getAll();
    public Sender getById(int id);
    public long count();
    public void insert(Sender sender);
    public void update(Sender sender);
    public void delete(Sender sender);
}
